src/modules/analytics/validators/price.candles.args.validator.ts

```diff
@@ -0,0 +1,109 @@
+import { ArgumentMetadata, Injectable, PipeTransform } from '@nestjs/common';
+import {
+    PriceCandlesQueryArgs,
+    PriceCandlesResolutions,
+} from '../models/query.args';
+import moment from 'moment';
+import { UserInputError } from '@nestjs/apollo';
+
+enum MaxMinutesPerResolution {
+    MINUTE_1 = 2880, // 2 days
+    MINUTE_5 = 4320, // 3 days
+    MINUTE_15 = 5760, // 4 days
+    MINUTE_30 = 10080, // 7 days
+    HOUR_1 = 14400, // 10 days
+    HOUR_4 = 21600, // 15 days
+    DAY_1 = 43200, // 30 days
+    DAY_7 = 172800, // 120 days
+    MONTH_1 = 1036800, // 2 years
+}
+
+const resolutionToMaxDuration = {
+    [PriceCandlesResolutions.MINUTE_1]: MaxMinutesPerResolution.MINUTE_1,
+    [PriceCandlesResolutions.MINUTE_5]: MaxMinutesPerResolution.MINUTE_5,
+    [PriceCandlesResolutions.MINUTE_15]: MaxMinutesPerResolution.MINUTE_15,
+    [PriceCandlesResolutions.MINUTE_30]: MaxMinutesPerResolution.MINUTE_30,
+    [PriceCandlesResolutions.HOUR_1]: MaxMinutesPerResolution.HOUR_1,
+    [PriceCandlesResolutions.HOUR_4]: MaxMinutesPerResolution.HOUR_4,
+    [PriceCandlesResolutions.DAY_1]: MaxMinutesPerResolution.DAY_1,
+    [PriceCandlesResolutions.DAY_7]: MaxMinutesPerResolution.DAY_7,
+    [PriceCandlesResolutions.MONTH_1]: MaxMinutesPerResolution.MONTH_1,
+};
+
+@Injectable()
+export class PriceCandlesArgsValidationPipe implements PipeTransform {
+    async transform(value: PriceCandlesQueryArgs, metadata: ArgumentMetadata) {
+        const { start, end, resolution } = value;
+
+        if (!this.isValidTimestamp(start)) {
+            throw new UserInputError('Invalid timestamp format for start.');
+        }
+
+        const startDate = moment.unix(parseInt(start));
+        const maxDuration = resolutionToMaxDuration[resolution];
```

Comment:
might check if `resolution` exists in the map

Reply:
there is no need for it. Graphql performs enum validation already